test/src/main/java/org/springframework/security/test/web/servlet/setup/SecurityMockMvcConfigurer.java

```diff
@@ -123,33 +123,47 @@ Filter getDelegate() {
 
 		@Override
 		public void init(FilterConfig filterConfig) throws ServletException {
+			this.requireDelegate();
 			this.delegate.init(filterConfig);
 		}
 
 		@Override
 		public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
 				throws IOException, ServletException {
+			this.requireDelegate();
 			this.delegate.doFilter(request, response, chain);
 		}
 
 		@Override
 		public void destroy() {
+			this.requireDelegate();
 			this.delegate.destroy();
 		}
 
 		@Override
 		public int hashCode() {
+			this.requireDelegate();
 			return this.delegate.hashCode();
 		}
 
 		@Override
 		public boolean equals(Object obj) {
+			this.requireDelegate();
 			return this.delegate.equals(obj);
 		}
 
 		@Override
 		public String toString() {
+			this.requireDelegate();
 			return this.delegate.toString();
 		}
+
+		private void requireDelegate() {
```

Comment:
Rather than do this could we update getDelegate to do a null check. Also keeping in mind that there is a race condition working on the member variable. So it would be something like:

```java
Filter result = this.delegate;
if (result == null) {
   // throw
}
return result;
```

Reply:
Force-pushed this modification, thanks!